### PR TITLE
Adds cURL support

### PR DESCRIPTION
**Pull request by NSMyself**
cURL support is a feature that Wormholy really should have in my honest opinion so I decided to add it myself.

I added tests and tried to follow Charles' output at all times. I mention this because there's more than one way of representing a request using cURL.

One drawback that I found was that Wormholy, unlike Charles, has no way of getting the User-Agent. As such, I was unable to add it to the cURL exporter. I'd love some feedback regarding possible implementations though.

### CODE REVIEW

File: Sources/UI/RequestDetailViewController.swift
Comment (NSMyself) [inline Sources/UI/RequestDetailViewController.swift:12]: Xcode automatically trimming trailing whitespaces

File: Wormholy.xcodeproj/project.pbxproj
Comment (NSMyself) [inline Wormholy.xcodeproj/project.pbxproj:303]: Xcode was failing while attempting to load the underlying module for the tests. Deleting the old test target and making a new one fixed it

File: WormholyTests/WormholyTests.swift
Comment (NSMyself) [inline WormholyTests/WormholyTests.swift:100]: I tried to add some of our actual endpoints in order to facilitate comparing with Charles' output; the actual payload i had to change though, since dictionaries are not ordered and comparing the output string would require so

File: Sources/Utils/RequestModelBeautifier.swift
Comment (NSMyself) [inline Sources/Utils/RequestModelBeautifier.swift]: POST requests in cURL do not need `-X POST` since it already contains `--data-binary`. Furthermore, JSON payload outputted by Charles used `--data-binary`, although [extensive documentation for cURL](https://ec.haxx.se/http-post.html) showed that `-d '{json}' -H 'Content-Type: application/json'` was enough.

As mentioned in the PR's description, I tried to follow Charles as closely as possible

**Review by andersio — DISMISSED** [review]

File: Sources/Utils/RequestModelBeautifier.swift
Comment (TheAdamBorek) [reply Sources/Utils/RequestModelBeautifier.swift]: Tbh I would still add `-X POST`. I didn't know this parameter is optional for POST and when I get some `cURL` I very often look what HTTPMethod it is.

I guess the reason why we want to have this `cURL` is to share with others. In that case I think it's better to have it (unless it's a big "code style" issue in cURL)


Although, If you decide to keep with the above solution, I think it would be nice to have a similar comment in the code. Ideally by exporting the `if` to a new private method and put the comment as function's documentation.

File: Sources/Utils/RequestModelBeautifier.swift
Comment (TheAdamBorek) [inline Sources/Utils/RequestModelBeautifier.swift]: do you need this variable?

File: WormholyTests/WormholyTests.swift
Comment (TheAdamBorek) [inline WormholyTests/WormholyTests.swift]: I think we shouldn't put those URLs here. This is a public fork and we have hidden URLs in the main app for some reason.

I think it's better to now have them here, as there are no benefits (it doesn't matter from test purpose if you test with babylon's URLs or Google's URL)

**Comment by TheAdamBorek** [comment]
I'm blocking the PR only because of public URLs. If we are fine with it I will dismiss the review but please double check on this

File: Sources/Utils/RequestModelBeautifier.swift
Comment (NSMyself) [reply Sources/Utils/RequestModelBeautifier.swift]: As mentioned, during development I chose to follow Charles' approach as much as possible since their export to cURL works well. This is simply a consequence of doing so.

Nevertheless we can still keep `-X POST`; personally I prefer having it there.

Comment (NSMyself) [reply Sources/Utils/RequestModelBeautifier.swift]: Done.

Comment (TheAdamBorek) [reply Sources/Utils/RequestModelBeautifier.swift]: I also prefer to have it there ;) 
I let you decide

**Review by TheAdamBorek — APPROVED** [review]

File: Sources/UI/RequestDetailViewController.swift
Comment (ilyapuchka) [inline Sources/UI/RequestDetailViewController.swift:64]: will it work on device? currently it's kind of painful to share requests from simulator or from device, I have to export in notes and make them use the same icloud account which feels so wrong...

File: Sources/Utils/RequestModelBeautifier.swift
Comment (ilyapuchka) [inline Sources/Utils/RequestModelBeautifier.swift:74]: this will only export request, right? maybe we can have it to export both request and curl and response in a separate file at the same time?

Comment (NSMyself) [reply Sources/Utils/RequestModelBeautifier.swift:74]: That's correct, just the request. The same way the other request models work, essentially.

File: Sources/UI/RequestDetailViewController.swift
Comment (NSMyself) [reply Sources/UI/RequestDetailViewController.swift:64]: It should, the same way it already exported the contents

File: Sources/Utils/RequestModelBeautifier.swift
Comment (NSMyself) [reply Sources/Utils/RequestModelBeautifier.swift:74]: @ilyapuchka ping ;)

File: WormholyTests/WormholyTests.swift
Comment (ilyapuchka) [inline WormholyTests/WormholyTests.swift:120]: I hope this one is not a real one =)

**Review by ilyapuchka — APPROVED** [review]

File: WormholyTests/WormholyTests.swift
Comment (NSMyself) [reply WormholyTests/WormholyTests.swift:120]: Nope 😊 but looking back I don't think I should have used this format precisely because of this: it will raise that question all the time. I'll change it to something a little less believable.

**Pull request closed without merging**
